### PR TITLE
feat: Decommission Legacy IaC Flow [CFG-1157]

### PR DESCRIPTION
**Pull request by YairZ101**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
Analytics show that `snyk iac test --legacy` has not been used by a non-snyk user for 90 days: https://snyk.grafana.net/d/zVy940uGx/iac-cli-ga-dashboard?orgId=1&from=now-7d&to=now&viewPanel=34

Therefore we have an opportunity to simplify our code by removing the legacy test code.

#### What are the relevant tickets?
[Jira ticket
](https://snyksec.atlassian.net/jira/software/c/projects/CFG/boards/301?modal=detail&selectedIssue=CFG-1157)

### CODE REVIEW

**Review by anthogez — APPROVED** [review]
Good job here @YairZ101!

Could you copy your GH's notes e.g.

```
Analytics show that snyk iac test --legacy has not been used by a non-snyk user for 90 days: https://snyk.grafana.net/d/zVy940uGx/iac-cli-ga-dashboard?orgId=1&from=now-7d&to=now&viewPanel=34

Therefore we have an opportunity to simplify our code by removing the legacy test code.
```
Also in your pull requests commit's message body?

Commit -> GitHub notes.

Thanks!

**Comment by YairZ101** [comment]
> Good job here @YairZ101!
> 
> Could you copy your GH's notes e.g.
> 
> ```
> Analytics show that snyk iac test --legacy has not been used by a non-snyk user for 90 days: https://snyk.grafana.net/d/zVy940uGx/iac-cli-ga-dashboard?orgId=1&from=now-7d&to=now&viewPanel=34
> 
> Therefore we have an opportunity to simplify our code by removing the legacy test code.
> ```
> 
> Also in your pull requests commit's message body?
> 
> Commit -> GitHub notes.
> 
> Thanks!

Thanks for reminding me. :)

File: src/cli/commands/test/iac-local-execution/assert-iac-options-flag.ts [hunk 1]
```diff
@@ -21,7 +21,6 @@ const keys: (keyof IaCTestFlags)[] = [
   'q',
   'quiet',
   'scan',
-  'legacy',
```
Comment (teodora-sandu) [inline src/cli/commands/test/iac-local-execution/assert-iac-options-flag.ts:24]: There's another `legacy` check at the bottom of the file btw:
```
export function isIacShareResultsOptions(options) {
  return options.iac && options.report && !options.legacy;
}
```

File: src/cli/commands/test/iac-local-execution/file-parser.ts [hunk 2]
```diff
@@ -161,3 +161,8 @@ export class UnsupportedFileTypeError extends CustomError {
     this.userMessage = `Unable to process the file with extension ${fileType}. Supported file extensions are tf, yml, yaml & json.\nMore information can be found by running \`snyk iac test --help\` or through our documentation:\nhttps://support.snyk.io/hc/en-us/articles/360012429477-Test-your-Kubernetes-files-with-our-CLI-tool\nhttps://support.snyk.io/hc/en-us/articles/360013723877-Test-your-Terraform-files-with-our-CLI-tool`;
   }
 }
+
+export function getFileType(filePath: string): string {
```
Comment (teodora-sandu) [inline src/cli/commands/test/iac-local-execution/file-parser.ts:165]: nit: Can we move this to `file-loader` instead? I understand why it's here (we check the `.fileType` field throughout this file) but the way I look at this is the loader prepares the files for the parser so it doesn't make sense for the function to be defined here 😅

**Review by teodora-sandu — APPROVED** [review]
Screenshot 2022-03-04 at 16 35 23

LGTM

File: src/cli/commands/test/iac-local-execution/file-parser.ts (hunk 2 above)
Comment (YairZ101) [reply src/cli/commands/test/iac-local-execution/file-parser.ts:165]: yea sure, I hope that after our CLI refactoring we will get rid of this function altogether and use [`path.extname`](https://attacomsian.com/blog/nodejs-get-file-extension) instead.

File: src/cli/commands/test/iac-local-execution/assert-iac-options-flag.ts (hunk 1 above)
Comment (YairZ101) [reply src/cli/commands/test/iac-local-execution/assert-iac-options-flag.ts:24]: funny that I missed the function that I created 1 week ago...
thx for catching that. :)

**Pull request merged**